src/Catalog/Helpers/AuditRecordHelpers.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Packaging.Core;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// Helper methods for accessing and parsing audit records.
+    /// </summary>
+    public static class AuditRecordHelpers
+    {
+        /// <summary>
+        /// Represents the information in the audit entry for a deletion.
+        /// </summary>
+        public class DeletionAuditEntry
+        {
+            /// <summary>
+            /// Over time, we have had three "deleted" file names.
+            /// </summary>
+            public static IList<string> FileNameSuffixes = new List<string>
+            {
+                "-Deleted.audit.v1.json",
+                "-deleted.audit.v1.json",
+                "-softdeleted.audit.v1.json",
+                "-softdelete.audit.v1.json",
+                "-delete.audit.v1.json"
+            };
+
+            public static async Task<DeletionAuditEntry> Create(Storage auditingStorage, Uri uri, CancellationToken cancellationToken, ILogger logger = null)
+            {
+                try
+                {
+                    return new DeletionAuditEntry(uri, await auditingStorage.LoadString(uri, cancellationToken));
+                }
+                catch (JsonReaderException)
+                {
+                    logger?.LogWarning("Audit record at {AuditRecordUri} contains invalid JSON.", uri);
+                    return null;
+                }
+                catch (NullReferenceException)
+                {
+                    logger?.LogWarning("Audit record at {AuditRecordUri} does not contain required JSON properties to perform a package delete.", uri);
+                    return null;
+                }
+            }
+
+            public DeletionAuditEntry(Uri uri, string contents)
+            {
+                Uri = uri;
+                Record = !string.IsNullOrEmpty(contents) ? JObject.Parse(contents) : null;
```

Comment:
This check doesn't seem to be required. If Record is null, InitValues will always throw. Therefore, null isn't even a valid value for Record, and we should probably throw before we even call InitValues.